src/store/reducers/createMapReducer/openMenuLevelsReducer.js

```diff
@@ -0,0 +1,33 @@
+import { TOGGLE_CATEGORIES, TOGGLE_GROUPS } from '../../constants/actionTypes';
+
+function toggleLevel(state, action) {
+  const { category, group, index, isRefresh } = action;
+  const level = category || group;
+
+  if (isRefresh) {
+    return [];
+  }
+
+  const openLevels = [...state];
+
+  if (index > -1) {
+    openLevels.splice(index, 1);
+  } else {
+    openLevels.push(level);
+  }
+  return openLevels;
+}
+
+export default function openMenuLevelsReducer(state = [], action) {
+  switch (action.type) {
+    case TOGGLE_CATEGORIES: {
+      return toggleLevel(state, action);
+    }
+    case TOGGLE_GROUPS: {
+      return toggleLevel(state, action);
+    }
+
+    default:
+      return state;
+  }
+}
```

Comment:
It would be really nice to include some docstrings here 🙏

Reply:
Sure...thanks.